### ISSUE TITLE
Battle test the OpenSearch Cluster with user defined args of API calls.

### ISSUE THREAD

**Issue by prudhvigodithi** [opensearch-project/opensearch-benchmark#342]
**Is your feature request related to a problem? Please describe.**

If would be nice to have an argument `opensearch-benchmark execute-test --api-calls [<LIST OF API CALLS TO TEST>]`, where benchmark can run tests against the user defined `--api-calls`. 

**Describe the solution you'd like**

A final score can be given to the `--api-calls` defined by the user.

@VachaShah @IanHoang

**Comment by IanHoang** [comment]
We'd have to incorporate and define each API in both the OSB and OSB workload repositories.

**Comment by gkamat** [comment]
@prudhvigodithi can you elaborate on the solution you would like to see?  OSB has the capability to issue an operation of the `raw-request` type, which permits any desired API call to be invoked; perhaps that might suffice.  It also permits operations to be run in parallel, if the schedule is so defined in the test procedure.

**Comment by rishabh6788** [comment]
OSB anyway calls the OS api under the hood like info, health, stats, search, index and a few others as it uses OpensearchPy client. 
What is the use-case of explicitly calling these api?

**Comment by prudhvigodithi** [comment]
Thanks @gkamat and @rishabh6788, can you please share some examples on how to run opensearch-benchmark tests on a user defined API's, example when I'm building and testing opensearch that has new API's part of the distribution, how to run opensearch-benchmark on that specific API and get the benchmark results?
@VachaShah

**Comment by rishabh6788** [comment]
OSB can only run the queries defined in the workload that is being used, for e.g., see https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/geonames/test_procedures/default.json for sample `geonames` workload. These queries are executed on an OS cluster using OpensearchPy client which internally invokes the associated APIs. 
If you want to run benchmark against a particular api then you can check if it is already supported in any of the available workloads, if not then you might want to add that by raising a PR. 
For e.g., k-NN team has a dedicated workload created that tests all the k-NN related queries/apis, see https://github.com/opensearch-project/k-NN/tree/main/benchmarks/osb. 
@gkamat Please correct me if I am wrong and there is any other way to achieve what @prudhvigodithi wants.

**Comment by VachaShah** [comment]
Thanks @rishabh6788! I connected with @rishabh6788 offline and he showed me how this can be done. I was able to try this out by modifying the workload default.json file.
